backend/app/services/posting_service.rb

```diff
@@ -65,14 +65,67 @@ def prefill(user:)
         applicant = Applicant.find_by(utorid: utorid)
         return { utorid: utorid } unless applicant
 
-        {
+        # This is the basic data we can get if the applicant already exists
+        data = {
             utorid: utorid,
             student_number: applicant.student_number,
             first_name: applicant.first_name,
             last_name: applicant.last_name,
             email: applicant.email,
             phone: applicant.phone
         }
+
+        existing_application = Application.find_by(posting: @posting)
```

Comment:
@alex-kozin This line turns out to be a huge bug...I wasn't finding the application by the applicant, I was just picking the first application for the posting...